### PR TITLE
Introduce common preset for Jest test configuration

### PR DESCRIPTION
**Pull request by literat**
## Description

We are repeating a lot of configuration for our tooling. There is a way to create presets for Jest and other tools. This is a first try that should present the preset for Jest, which can be used by all packages.

There will be some future steps that will introduce the usage of setup files and general mocking. This PR is now moving the most common configurations into the preset. It tries not to break everything.

There is also a move of the `jest.config` file from the `config` directory to the root of each package. Please consider if this is a good approach, but it seems to me, that all packages across the JS-verse are now adopting the similar naming of the configuration file and that is `*.config.{js, ts, mjs, cjs}`. Actually, I think it is a good move but what I am missing now are the files starting with the dot `.`, because they are naturally sorted by name, and I know that all the files with the starting dot are configurations. Now with the `jest.config` and `eslint.config` and `prettier.config` it will be really messy there. 🤷‍♂️

Note: there are actually two presets, one for the browser environment and one for the server. The server is used by default. Please, see a readme of the package.

### Additional context

- https://github.com/carbon-design-system/carbon/blob/e4922a44b2110b9dcb71da21ff6c0c9bb95610e0/config/jest-config-carbon/index.js
- https://jestjs.io/docs/configuration#preset-string

### CODE REVIEW

File: configs/jest-config-spirit/jsdom/setup/setupAfterEnv.js
Comment (literat) [inline configs/jest-config-spirit/jsdom/setup/setupAfterEnv.js:1]: Note to myself: Move the commented code out of this or add some note about that the code should replace the one in the `web-react` package setup file.

File: configs/jest-config-spirit/node/index.js [hunk 1]
```diff
@@ -14,9 +16,15 @@ const config = {
...
   // https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object
   transform: {
-    '^.+\\.(t|j)s?$': ['<rootDir>/../../node_modules/@swc/jest'],
+    '^.+\\.(t|j)sx?$': [resolve('./../../node_modules/@swc/jest')],
   },
 
+  // This line is because of polyfill for useResizeHook
+  // An array of regexp pattern strings that are matched against all source file paths before transformation.
+  // https://jestjs.io/docs/configuration#transformignorepatterns-arraystring
+  // transformIgnorePatterns: [require.resolve('./../../node_modules/@juggle/resize-observer')],
```
Comment (literat) [inline configs/jest-config-spirit/node/index.js:25]: Remove this line.

File: packages/web-react/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const config = {
+  preset: 'jest-config-spirit/jsdom',
+
+  // A list of paths to modules that run some code to configure or set up the testing framework before each test file in the suite is executed.
+  // https://jestjs.io/docs/configuration#setupfilesafterenv-array
+  setupFilesAfterEnv: ['<rootDir>/config/jest/setupTestingLibrary.ts'],
```
Comment (literat) [inline packages/web-react/jest.config.ts:6]: Note to myself: Add a comment about that this should be moved to the `jest-config-spirit`.

File: configs/jest-config-spirit/node/index.js (hunk 1 above)
Comment (literat) [reply configs/jest-config-spirit/node/index.js]: Fixed.

File: packages/web-react/jest.config.ts (hunk 2 above)
Comment (literat) [reply packages/web-react/jest.config.ts:11]: Comment added.

File: configs/jest-config-spirit/jsdom/setup/setupAfterEnv.js
Comment (literat) [reply configs/jest-config-spirit/jsdom/setup/setupAfterEnv.js:1]: Fixed.

File: configs/jest-config-spirit/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "private": true,
+  "description": "Jest configuration and preset for Spirit Design System",
+  "version": "0.0.0",
+  "license": "MIT",
+  "keywords": [
+    "lmc-eu",
+    "spirit",
+    "spirit-design-system",
+    "components",
+    "react"
```
Comment (crishpeen) [inline configs/jest-config-spirit/package.json:12]: are these correct?

File: configs/jest-config-spirit/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# jest-config-spirit
+
+> Jest configuration and preset for Spirit Design System
+
+## Getting started
+
+This preset is part of this monorepo only and thus it is accessible for every package.
+
+To install `jest-config-carbon` in your project, you will need to run the following command using [Yarn](https://yarnpkg.com/en/):
```
Comment (crishpeen) [inline configs/jest-config-spirit/README.md:9]: I see you were inspired somewhere :D

Comment (crishpeen) [reply configs/jest-config-spirit/README.md:9]:
```suggestion
To install `jest-config-spirit` in your project, you will need to run the following command using [Yarn](https://yarnpkg.com/en/):
```

Comment (crishpeen) [reply configs/jest-config-spirit/README.md:9]: And you might consider moving links to the bottom of the file as we do it elsewhere :)

**Review by crishpeen — APPROVED** [review]
LGTM

File: configs/jest-config-spirit/README.md (hunk 4 above)
Comment (literat) [reply configs/jest-config-spirit/README.md]: Fixed and moved.

**Pull request merged**